pkg/ingester/errors.go

```diff
@@ -26,36 +28,62 @@ var (
 		"the ingester is currently too busy to process queries, try again later")
 )
 
+// ErrorWithStatus is used for wrapping errors returned by ingester.
```

Comment:
We can keep this private to ingester. Distributor will never encounter this type, so we don't need to test for it.